recipes/qpdf/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "5",
+            "Visual Studio": "19.10",
+            "clang": "3.4",
+            "apple-clang": "10",
+        }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC # once removed by config_options, need try..except for a second del
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src") # src_folder must use the same source folder name the project
+
+    def requirements(self):
+        # https://qpdf.readthedocs.io/en/stable/installation.html#basic-dependencies
+        self.requires("zlib/1.2.12")
+        self.requires("libjpeg-turbo/2.1.4")
+
+
+    def validate(self):
+        # validate the minimum cpp standard supported. For C++ projects only
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+        # in case it does not work in another configuration, it should validated here too
+        if is_msvc(self) and self.info.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    # if another tool than the compiler or CMake is required to build the project (pkgconf, bison, flex etc)
+    def build_requirements(self):
+        self.tool_requires("cmake/3.24.1")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        # BUILD_SHARED_LIBS and POSITION_INDEPENDENT_CODE are automatically parsed when self.options.shared or self.options.fPIC exist
+        tc = CMakeToolchain(self)
+        # Boolean values are preferred instead of "ON"/"OFF"
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.variables["BUILD_STATIC_LIBS"] = not self.options.shared
+        # for now: enforce built-in crypto easing up the crypto dependency management
+        tc.variables["USE_IMPLICIT_CRYPTO"] = False
+        tc.variables["REQUIRE_CRYPTO_NATIVE"] = True
+        tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+        tc.variables["REQUIRE_CRYPTO_OPENSSL"] = False
+        if is_msvc(self):
+            # don't use self.settings.compiler.runtime
+            tc.cache_variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = not is_msvc_static_runtime(self)
```

Comment:
That was just part of the template. I don't have an MSVC at my hand, so i can't confirm if this is necessary. When you doubt it yourself, then i will just remove it as long as no test and no bugreport will state otherwise ;)